### PR TITLE
treewide: switch to fetchFromGitHub

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
use `fetchFromGitHub`

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/box2d/default.nix [hunk 1]
```diff
@@ -22,6 +24,7 @@ stdenv.mkDerivation rec {
   ];
 
   prePatch = ''
+    find
```
Comment (veprbl) [inline pkgs/development/libraries/box2d/default.nix:27]:
```suggestion
```

Comment (Stunkymonkey) [reply pkgs/development/libraries/box2d/default.nix]: thanks for noticing. stupid mistake

Comment (veprbl) [reply pkgs/development/libraries/box2d/default.nix]: It's fine, good to see that this was tested.

Comment (veprbl) [reply pkgs/development/libraries/box2d/default.nix]: It's fine, good to see that this was tested.

**Comment by Stunkymonkey** [comment]
@ofborg eval

File: pkgs/development/libraries/audio/libsmf/default.nix [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
   version = "1.3";
   pname = "libsmf";
-  src = fetchurl {
-    url = "https://github.com/stump/libsmf/archive/${pname}-${version}.tar.gz";
-    sha256 = "1527pcc1vd0l5iks2yw8m0bymcrnih2md5465lwpzw0wgy4rky7n";
+
+  src = fetchFromGitHub {
+    owner = "stump";
+    repo = "libsmf";
+    rev = "libsmf-1.3";
```
Comment (bobby285271) [inline pkgs/development/libraries/audio/libsmf/default.nix:10]:
```suggestion
    rev = "libsmf-${version)";
```

**Comment by Stunkymonkey** [comment]
@veprbl is it possible to get it merged into `master`?

**Comment by veprbl** [comment]
Sure. Would be nice to run nixpkgs-review on this.

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 152297` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>box2d</li>
    <li>capstone</li>
    <li>cmrt</li>
    <li>libsForQt5.grantlee</li>
    <li>libsmf</li>
    <li>lvtk</li>
    <li>openbabel2</li>
  </ul>
</details>

**Review by veprbl — APPROVED** [review]

**Pull request merged**